core/src/main/java/org/apache/calcite/rel/rules/FilterJoinRule.java

```diff
@@ -53,28 +52,21 @@
    * will be pushed into the ON clause. */
   public static final Predicate TRUE_PREDICATE = (join, joinType, exp) -> true;
 
-  /** Predicate that returns true if the join is not Enumerable convention,
-   * will be replaced by {@link #TRUE_PREDICATE} once enumerable join supports
-   * non-equi join. */
-  // to be removed before 1.22.0
-  private static final Predicate NOT_ENUMERABLE = (join, joinType, exp) ->
-      join.getConvention() != EnumerableConvention.INSTANCE;
-
   /** Rule that pushes predicates from a Filter into the Join below them. */
   public static final FilterJoinRule FILTER_ON_JOIN =
       new FilterIntoJoinRule(true, RelFactories.LOGICAL_BUILDER,
-          NOT_ENUMERABLE);
+          TRUE_PREDICATE);
 
```

Comment:
If NOT_ENUMERABLE is not needed any more, should we also no need to retain the predicate parameter?

Reply:
I guess still need it for backward compatibility, in case some other is still using it.